yoyodyne/collators.py

```diff
@@ -16,23 +20,70 @@ class Collator:
     has_features: bool
     has_target: bool
     separate_features: bool
+    max_source_length: int
+    max_target_length: int
 
-    def __init__(self, pad_idx, config: dataconfig.DataConfig, arch: str):
+    def __init__(
+        self,
+        pad_idx,
+        config: dataconfig.DataConfig,
+        arch: str,
+        max_source_length: int,
```

Comment:
Do you want to pass in sensible defaults here? They are both 128 at the flag level, I believe, which seems like a decent default.

Reply:
I never liked setting defaults in two different places. But since this could potentially be used as a CLI or a python library -- I have no idea what is typical here. I guess defaults here does make the most sense.

Reply:
I don't love this either, but we could have `defaults.py` which could contain these values as constants. What do you think? (That would be a separate PR.)

(So far I have just repeated ourselves in most places since there are other reasons to have defaults, e.g., for testing, and I want this library to be very useable from Python too.)

Reply:
`defaults.py` sounds like a good solution.